lib/ui/recommendation.js

```diff
@@ -61,7 +59,7 @@ Recommendation.prototype = {
   navigate: function(evt) {
     // if it's a click, we'll have a MouseEvent; if it's a right-click, bail.
     // else, we'll just have the (possibly null) data from the 'enter-key' event.
-    // call this.urlbar.navigate to trigger navigation, passing in the URL.
-    // TODO: or should we fire an event and keep all the modules maximally disconnected?
+    // fire a 'navigate-requested' event (please find a better name) to trigger navigation.
```

Comment:
Thinking about event names. Let's use the form `prefix-verb` with three different prefixes:

- `urlbar` for any events in the urlbar itself.
- `recommendation` for any events related to our hbox
- `awesomebar` for any events related to the awesomebar that are not our hbox 

So here, maybe `recommendation-navigate`?